### ISSUE TITLE
How to modify Discon.sln files and other module files to create a wind turbine with a rated wind speed of 10m/s and a rated power of 2.2MW in OpenFAST?

### ISSUE THREAD

**Issue by 1codecopyer** [OpenFAST/openfast#2499]
**The rated wind speed of the 2.2MW wind turbine model built by OpenFAST does not match the actual wind speed.**
The blue scatter represents the wind speed power data of the 2.2MW wind turbine recorded by SCADA, and the red scatter represents the simulated wind speed power curve of the 2.2MW wind turbine created by OpenFAST. The rated wind speed of the simulated 2.2MW wind turbine is greater than 10m/s, which means that the simulated 2.2MW wind turbine has a power of about 1450kW at 10m/s, which does not reach the expected 2200kW!
![Actual SCADA data V S  OpenFAST Simulation Power Result](https://github.com/user-attachments/assets/0452b420-65b9-4ea0-86bc-5bcc86075cbf)

**To Reproduce**

Based on the NREL 5MW Baseline wind turbine model, module parameters are modified to create a 2.2MW wind turbine model in OpenFAST. The modified parameters include the rotor radius in the ED module, blade length in the AD module, inflow wind speed in the InflowWind module, and the DLL controller in the ServoDyn module. Please refer to the image for specific details of the changes.

The version running information of OpenFAST is shown in the figure:
![OpenFAST run info](https://github.com/user-attachments/assets/4d269e81-1237-45b4-a70e-7722ebf6788b)

In the simulation main file fst, the total simulation duration is set to 120 seconds and the simulation step size is 0.01 seconds. The main module input files are shown in the figure:
![2 2 MW WT main fst file input files parameters](https://github.com/user-attachments/assets/9e86d14d-026a-4e88-b5bf-c2274a2d2a94)

The initial speed of the rotor is set to 10rpm.The tower heights of the ED module and AD module are kept consistent, and the details of parameter changes in the input files of the ED and AD modules are shown in the following figures:
![2 2MW WT ED Parameters](https://github.com/user-attachments/assets/35029b3a-c3e2-4705-a2bd-5fee285eb3e6)

The wing profile file of AD is consistent with the NREL 5 MW baseline
![2 2MW WT AD Parameters](https://github.com/user-attachments/assets/669a61f2-dab8-4001-98de-fc015e4aad98)

At the same time, the blade length of the AD module is consistent with the wind turbine radius of 59 meters in the ED module
![2 2 MW WT AD Blade Parameters](https://github.com/user-attachments/assets/80eef1f2-a5dc-4a74-b08d-d3ff9fe515cd)

Set the wind type of OpenFAST to steady-state wind and set the wind speed to 10m/s, which is consistent with the rated wind speed of a real 2.2MW wind turbine. Check if the 2.2MW wind turbine model built by OpenFAST outputs 2.2MW power at 10m/s The parameters of the InflowWind module are shown in the figure:
![2 2 MW WT InflowWind Parameters](https://github.com/user-attachments/assets/93e95545-3c5c-4af4-8e73-4e010abb19ec)
![2 2 MW WT ServoDyn Bladed Interface(Bladed-style DLL)](https://github.com/user-attachments/assets/7f12df81-44b3-406f-8741-30208319b7c0)

The ServoDyn module uses the Bladed style dll controller interface to match with the 2.2MW wind turbine, as shown in the figure:
![2 2 MW WT ServoDyn PCmode and VSControl Parameters](https://github.com/user-attachments/assets/5df9416f-bc8e-4a59-9d17-d61fad7b8a0b)

Because modifications need to be made to the 5MW Baseline fan controller:
The basis for the modification is the response curve of the actual torque and speed of the 2.2MW generator, as shown in the figure:
![image](https://github.com/user-attachments/assets/f37746a8-8711-403a-9ef7-2ab9251278e3)
![image](https://github.com/user-attachments/assets/3417cd07-913b-4590-b892-5b5e1afc3d36)
Based on the above two pictures, the key speeds of the actual 2.2MW wind turbine generator can be obtained by analogy, which are:
ωA=750 rpm (Region 1)
ωB=1080 rpm (Region 1 1/2)
ωC=1700 rpm (Region 2 1/2)
ωD=ωg,rated=1750 rpm (Region 3)
VS_RtTq=12250Nm
Based on the key speed of the generator mentioned above, modify the Fortran source code of DISCON.sln in the vs build folder of OpenFAST, and generate a solution to control the 2.2MW wind turbine by generating a DLL file. The modified parameters of the controller are shown in the figure:
![Parameters correction in VS_Build DISCON sln  for 2 2MW onshore WT](https://github.com/user-attachments/assets/6c25fa37-5559-4460-8bdd-196c8d207be3)

By drawing the result file of OpenFAST operation, the power data and simulation time curve of the 2.2MW wind turbine model built at 10m/s were plotted. It was found that the output power of the 2.2MW wind turbine with a rated wind speed of 10m/s built in OpenFAST was only 1450kW at 10m/s, which did not reach the expected 2200kW. The result is shown in the figure:
![OpenFAST simulation Power at 10 mps wind speed](https://github.com/user-attachments/assets/acdd3f45-b2a7-47ff-a685-814bd04f2af9)

**Expected behavior**
<Build a wind turbine model with a rated wind speed of 10m/s and a rated power of 2.2MW in OpenFAST>

**Additional context**
<Add any other context about the problem here.>

**Comment by jjonkman** [comment]
Dear @1codecopyer,

As far as I can tell, you are trying to model a 2.2 MW wind turbine by making small changes to the OpenFAST model of the NREL 5-MW baseline wind turbine supplied by NREL, including truncating the blade radius from 63 to 59 m and changing constants within the DISCON controller.  I wouldn't expect this to be a great representation of a specific 2.2 MW turbine, but perhaps it is an OK starting point.

Looking briefly at the data you shared, I see one obvious mistake: you've set `NumBlNds` = 13 in the AeroDyn blade input file, meaning that your aerodynamic discretization will only have 13 nodes, extending from the root (0.0) to 43 m, and all data in your blade file beyond row 13 is ignored.  So, aerodynamically, your radius is much less than 59 m.  I'd change `NumBlNds` to 18 and BlSpn(18) to 59 - 1.5 = 57.5 m and try again.

Best regards,

**Comment by 1codecopyer** [comment]
I looked closely at the NREL 5 MW Baseline RWT input file, where the `Tiprad` is 63 m, taking into account the hub radius `HubRad` of 1.5 m, so the actual blade length in the AD module is 63-1.5 = 61.5 meters. Similarly, I should modify the blades of the 2.2 MW WT to 59-1.5 = 57.5 m and to be consistent with `NumBlNds`. This solves the input parameter problem of the pneumatic module. Then, by analyzing the SCADA data of the actual 2.2 MW WT, using the generator speed and rotor speed, I modified the gearbox `GBRatio` of the drive system so that when the rated power is reached, the rotor speed of the wind turbine and the speed of the generator are consistent with the actual, basically satisfying the simple 2.2 MW single wind turbine model.
Thanks to @jjonkman
